### ISSUE TITLE
ninja: build stopped: subcommand failed.

### ISSUE THREAD

**Issue by ljq2020** [autonomousvision/occupancy_networks#96]
Hi,

When I run `python setup.py build_ext --inplace`, I met a problem
![3](https://user-images.githubusercontent.com/63891270/113418245-ad61b900-93f7-11eb-9a0c-d867cda2794d.png)
![4](https://user-images.githubusercontent.com/63891270/113418254-b18dd680-93f7-11eb-83a3-f2f25cacb8cd.png)

Does anyone know how to solve it? Thanks in advance.

**Comment by AlexsaseXie** [comment]
I have the same problem with the RTX 3090 server which use CUDA 11.1.
So the correspoding pytorch has to be 1.8.1 (or newer).
By the way, g++ version is 9.3.0.

**Comment by tsunghan-mama** [comment]
I have the same problem QQ.

**Comment by AlexsaseXie** [comment]
OK... Finally I fixed this issue.

**It is because the building error of pykdtree which is written in C (not C++).** 
I don't clearly know the mechanism of pytorch but I found that in pytorch 1.0.0, the BuildExtension module will call gcc to compile the C code which should be Okay. While in pytorch 1.8.1, the BuildExtension module will call c++ to compile which will cause error because of the type checking or some other grammar issue.

So the solution is to write another setup_c.py to compile the pykdtree module which uses standard build_ext module.
The code is like follows:
`
…
from setuptools.command.build_ext import build_ext
…
ext_modules = [
    pykdtree,
]

setup(
    ext_modules = cythonize(ext_modules),
    include_dirs=[numpy.get_include()],
    cmd_class={
        'build_ext': build_ext
    }
)
`[python]
